### PR TITLE
prevent permit full loads from closed download modal, emails from not…

### PR DESCRIPTION
**Pull request by bkmaibach**
… previously nil viewed_at

## Description

fix download modal from downloading full permit application thus setting viewed_at when its closed.

and only send the email when viewed_at was previously nil.

### CODE REVIEW

File: app/models/permit_application.rb [hunk 1]
```diff
@@ -209,10 +209,12 @@ def take_form_customizations_snapshot_if_submitted
 
   def notify_user_application_updated
     return if new_record?
-
-    return unless saved_change_to_viewed_at? || saved_change_to_reference_number?
-
-    PermitHubMailer.notify_application_updated(submitter, self).deliver_later
+    viewed_at_change = previous_changes.dig("viewed_at")
+    binding.pry
```
Comment (taw27) [inline app/models/permit_application.rb:213]: binding.pry

File: app/models/permit_application.rb [hunk 2]
```diff
@@ -209,10 +209,12 @@ def take_form_customizations_snapshot_if_submitted
 
   def notify_user_application_updated
     return if new_record?
-
-    return unless saved_change_to_viewed_at? || saved_change_to_reference_number?
-
-    PermitHubMailer.notify_application_updated(submitter, self).deliver_later
+    viewed_at_change = previous_changes.dig("viewed_at")
```
Comment (taw27) [inline app/models/permit_application.rb:212]: Is it possible to use this api `attribute_changed?(attr_name)` , I believe it should give a boolean for if the attribute was changed

Comment (taw27) [reply app/models/permit_application.rb:212]: if not good to go

**Review by taw27 — CHANGES_REQUESTED** [review]

File: app/models/permit_application.rb (hunk 2 above)
Comment (bkmaibach) [reply app/models/permit_application.rb:212]: this is the only way I could get it to work. I'm trying to find not when it was changed but specifically when it was changed from nil

**Review by taw27 — APPROVED** [review]

**Pull request merged**